### DISCUSSION TITLE
Inject value from vscode into Notebook variable

### DISCUSSION THREAD

**Discussion by vaclavHala** [microsoft/vscode-discussions#895] [Extension Development QnA] [answered]
I would like to be able to inject some values that live in vscode (e.g. user configurations, or result of custom command provided by some extension ) into notebook variable. It would be great if something like this were possible directly in any code cell
```python
model_dir = ${myExtension.currentModelDir} # this calls given command and assigns its string return value to model_dir
print(model_dir)
```
But if that is too much to ask it would also be fine to write an extension which can interpret some special type of cell, something like this
```json
{
    "cell_type": "code",
    "metadata": {
        "vscode": {
            "languageId": "commandCaller"
        }
    },
    "outputs": [],
    "source": [
        "model_dir = myExtension.currentModelDir"
    ]
}
```
which interprets cells with simple syntax `$VAR = $COMMAND` by calling the command by name and assigning its output to given variable that can then be referenced later in regular code cells.

Looking at the Notebook API I can see no way to do this, in fact I see no mention of variables at all which makes me think they are sort of "internal implementation detail" of the Jupyter execution engine that can't be accessed from the outside at all?

**Comment by theReynald** [comment, marked as answer]
Hi @vaclavHala ,

This is kind of a feature request so I'd say to file an issue here https://github.com/microsoft/vscode-jupyter/issues
